frontend/src/views/SoundEditingUpload.vue

```diff
@@ -0,0 +1,286 @@
+    this.files.forEach(({ uploadRemainingTime: uploadTime = 0 }) => {
+      totalUploadTime += uploadTime === Infinity ? 0 : uploadTime;
+    });
+    return totalUploadTime < 60
+      ? "Less than a minute left"
+      : moment.duration(totalUploadTime, "seconds").humanize();
+  }
+
+  uploadFile(path: string, file: File) {
+    const metadata = {
+      customMetadata: {
+        fullName: file.name
+      }
+    };
+    const uploadTask = fb
+      .storage()
+      .ref()
+      .child(path)
+      .put(file, metadata);
+    this.updateFileFields(file, {
+      uploading: true,
+      uploadTask: uploadTask,
+      uploadStartedAt: new Date()
+    });
+    uploadTask.on(
+      "state_changed",
+      ({ bytesTransferred, totalBytes }: any) => {
+        const { uploadStartedAt } = this.getFile(file);
+        const startedAt = uploadStartedAt ? uploadStartedAt.getTime() : 0;
+        const secondsElapsed = (new Date().getTime() - startedAt) / 1000;
+        const bytesPerSecond = secondsElapsed
+          ? bytesTransferred / secondsElapsed
+          : 0;
+        const remainingBytes = totalBytes - bytesTransferred;
+        const progress = (bytesTransferred / totalBytes) * 100;
+        this.updateFileFields(file, {
+          progress: Math.round(progress),
+          uploadRemainingTime: secondsElapsed
+            ? remainingBytes / bytesPerSecond
+            : 0
+        });
+      },
+      (error: any) => {
+        if (error.code === "storage/canceled") {
+          this.updateFileFields(file, {
+            error: "Canceled upload",
+            complete: false,
+            uploading: false
+          });
+        } else if (error.code === "storage/unauthorized") {
+          this.updateFileFields(file, {
+            error: "You are not authorized to upload this file",
+            complete: false,
+            uploading: false
+          });
+        } else {
+          this.emitFileError(file, error.message);
+        }
+      },
+      async () => {
+        this.completedFileUploads += 1;
+        this.updateFileFields(file, {
+          complete: true,
+          uploading: false
+        });
+      }
+    );
+  }
+
+  template() {
+    return `<div class="dz-preview dz-file-preview fileWrapper">
+                <div class="dz-image">
+                    <div data-dz-thumbnail-bg></div>
+                </div>
+                <div class="dz-details">
+                    <div class="dz-size"><span data-dz-size></span></div>
+                    <div class="dz-filename"><span data-dz-name></span></div>
+                </div>
+                <div class="dz-progress"><span class="dz-upload" data-dz-uploadprogress></span></div>
+                <div class="dz-error-message"><span data-dz-errormessage></span></div>
+                <div class="dz-success-mark"><i class="fa fa-check"></i></div>
+                <div class="dz-error-mark"><i class="fa fa-close"></i></div>
+            </div>
+        `;
```

Comment:
Is it necessary to declare the template as a string?

Reply:
Since it's being hidden and the uploaded files displayed below it has a custom CSS class to make sure other styles don't affect it.

Reply:
Still not clear but I'm leaving it up to you. Just wondered.